maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -320,6 +324,49 @@ void initialize(CliRequest cliRequest) throws ExitException {
             }
         }
 
+        Path topDirectory = Paths.get(cliRequest.workingDirectory);
+        boolean isAltFile = false;
+        for (String arg : cliRequest.args) {
+            if (isAltFile) {
+                Path path = Paths.get(arg);
+                if (Files.isDirectory(path)) {
+                    topDirectory = path;
+                } else if (Files.isRegularFile(topDirectory)) {
+                    topDirectory = path.getParent();
+                    if (!Files.isDirectory(topDirectory)) {
+                        System.err.println("Directory " + topDirectory
+                                + " extracted from the -f/--file command-line argument " + arg + " does not exist");
+                        throw new ExitException(1);
+                    }
+                } else {
+                    System.err.println(
+                            "POM file " + arg + " specified with the -f/--file command line argument does not exist");
+                    throw new ExitException(1);
+                }
+                break;
+            } else {
+                isAltFile = arg.equals(String.valueOf(CLIManager.ALTERNATE_POM_FILE)) || arg.equals("file");
```

Comment:
This looks pretty low level, can't we use high level CLI to get this value?

Reply:
Well, the CLI isn't built yet, this is really one of the first things done.  Also the point is to be able to leverage those properties during argument interpolation (see https://github.com/apache/maven/pull/1062), so this really has to be done very early.

Reply:
Aha, then a comment should accompany that code,

Reply:
Added some comments.